include/sound/sof/topology.h

```diff
@@ -174,6 +174,22 @@ struct sof_ipc_comp_asrc {
 	uint32_t reserved[4];
 } __attribute__((packed));
 
+struct mux_stream_data {
+	uint32_t data_id;
+	uint32_t pipeline_id;
+	uint8_t num_channels;
+	uint8_t mask[8];
+	uint8_t reserved[3]; /* padding */
+} __packed;
```

Comment:
I don't think we have u8 in IPC definitions? Are you sure we're not going to have endianness issues or other shenanigangs?

Reply:
no I'm not. Actually the 8 and 3 in the mask and reserved are also dubious as we have variable size for those in FW depending on platform (for e.g. 8 for apl, 4 for byt). So actually I would like us also to rationalize the mux/demux structs, if they are made public in kernel, so that it would be same for all platforms and have some sane types. I mean pipeline_id is uint32 and channels is uint8? Are we going to have 2 million pipelines? So why don't just set everything to uint16 and fix the max channels to 8?

I think the space save we get from these hacks is minimal. So this is more for mux/demux component developer, but as these structs could be mirrored in kernel, they should take into account other issues as minimal size also...